product-service/src/dataProvider.ts

```diff
@@ -1,10 +1,74 @@
 import data from './data';
+import { Client } from 'pg';
+import { Pool } from 'pg';
+
+const { PG_HOST, PG_PORT, PG_DATABASE, PG_USERNAME, PG_PASSWORD } = process.env;
+const dbOptions = {
+    host: PG_HOST,
+    port: PG_PORT,
+    database: PG_DATABASE,
+    user: PG_USERNAME,
+    password: PG_PASSWORD,
+    ssl: {
+        rejectUnauthorized: false,
+    },
+    connectionTimeoutMillis: 5000,
+};
 
 export const getProducts: any = async () => {
-  return Promise.resolve(data);
+  const client = new Client(dbOptions);
+  await client.connect();
+  console.log('Connected to DB');
+
+  const ddlResult = await client.query(`
+  SELECT * FROM products LEFT JOIN stocks ON products.id = stocks.product_id`);
+
+  console.log('Fetched products from DB - ', ddlResult.rows);
+
+  return Promise.resolve(ddlResult.rows);
 }
 
 export const getProductItemById: any = async (id) => {
-  const product = data.find(item => item.id === +id);
-  return Promise.resolve(product);
+  const client = new Client(dbOptions);
+  await client.connect();
+  console.log('Connected to DB');
+
+  const ddlResult = await client.query(`
+  SELECT * FROM products WHERE id = '${id}'`);
```

Comment:
it's better to use `SELECT * FROM products WHERE id = $1` to avoid [sql injections](https://node-postgres.com/features/queries)

Reply:
Fixed